src/wallet/wallet.cpp

```diff
@@ -2251,6 +2250,7 @@ bool CWallet::SetHDSeed(const HDSeed& seed)
 
     {
         LOCK(cs_wallet);
+        CWalletDB(strWalletFile).WriteNetworkInfo(networkIdString);
```

Comment:
AFAICT this will only be called when `GenerateNewSeed` is called (i.e. for brand new wallets that don't have an `hdseed`), so existing wallets would not store this value, and thus will have no entry to check their configured parameters against. We have two options:

- Keep the PR as-is. Existing wallets will on every launch assume that the `wallet.dat` is for the network corresponding to the datadir the wallet was loaded from, unless `-wallet=/absolute/path` is given, in which case we're assuming the wallet is for the network the node was started from. This matches existing behaviour, and continually trusts the node operator (or e.g. ZecWallet Full Node user) to configure their node correctly.
- Persist this information for existing wallets if it is missing. This is effectively a "Trust On First Use" model: the above assumption is made the first time the node is restarted, and from that point onwards the `wallet.dat` tracks the network. If the node operator makes a mistake on that first restart, that will persist and (currently) there is no way to override it. But assuming no mistake is made, the wallet is protected going forward.

Reply:
For this PR it should be left as-is, because the introduction of mnemonic seeds will change the paths by which this is reachable - the associated network information will be persisted when the wallet is upgraded with the new mnemonic seed, so it will begin behaving as "Trust on first use" for existing wallets at that point.